pkgs/top-level/all-packages.nix

```diff
@@ -37493,6 +37493,45 @@ with pkgs;
 
   kakasi = callPackage ../tools/text/kakasi { };
 
+  ### GAMES/XASH3D
+
+  xash3d-unwrapped = callPackage ../games/xash3d { };
+
+  hlsdk-valve = callPackage ../games/xash3d/hlsdk.nix {
+    name = "valve";
+    version = "unstable-2023-07-17";
+    rev = "8180cf60c270b933096506f61d07ed03ba3619af";
+    sha256 = "sha256-HPPsP2Y8LQFbMDek//xrsKaDULhDsdeTQyZYDnmOzt0=";
+  };
+  xash3d-valve = callPackage ../games/xash3d/wrapper.nix {
+    hlsdk = hlsdk-valve;
+    name = "valve";
+    dllname = "hl";
+  };
+
+  hlsdk-bshift = callPackage ../games/xash3d/hlsdk.nix {
+    name = "bshift";
+    version = "unstable-2023-07-17";
+    rev = "818632f2baf2cf610c7fadbc2ec1ad906f1a8f29";
+    sha256 = "sha256-UymPWgiQKxZnPuoTCPdAaCMLmlZEjlr1aA5x+oYW6LI=";
+  };
+  xash3d-bshift = callPackage ../games/xash3d/wrapper.nix {
+    hlsdk = hlsdk-bshift;
+    name = "bshift";
+  };
+
+  hlsdk-gearbox = callPackage ../games/xash3d/hlsdk.nix {
+    name = "gearbox";
+    version = "unstable-2023-07-17";
+    rev = "ee125707e2185d8bf3f35f8da67fafa893769c4f";
+    sha256 = "sha256-w4Gg5kTB6yUj+BZxDMLMEksMBAdjQlywHuFAROdTZBo=";
+  };
+  xash3d-gearbox = callPackage ../games/xash3d/wrapper.nix {
+    hlsdk = hlsdk-gearbox;
+    name = "gearbox";
+    dllname = "opfor";
+  };
```

Comment:
Please move these into games/xash3d/default.nix and use callPackages here

Reply:
Well this was exactly the thing for which I was asking for help :p I had no clue how to do it. But finally I stumbled upon quake2 which showed me a good way. I now have a file `games.nix` where they are all defined, and with a function to avoid duplication, just the way I wanted to do it. In the future I will easily be able to add new games, which is good because there are many others which I didn't package yet. I am first trying to get this stuff merged, and besides I haven't tested any of the other games and am not interested in doing so right now.